cc_unitsearch.py

```diff
@@ -0,0 +1,86 @@
+import sys
+import json
+
+from PyQt5.QtWidgets import QApplication, QWidget, QLineEdit, QTableView, QHeaderView, QVBoxLayout
+from PyQt5.QtCore import Qt, QSortFilterProxyModel, QRegExp
+from PyQt5.QtGui import QStandardItemModel, QStandardItem
+
+def pullFromJson():
+    with open('CC_UNITS.json') as json_file:
+        data = json.load(json_file)
+        
+        return data
+
+class CustomFilter(QSortFilterProxyModel):
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.role = Qt.DisplayRole
+
+    def filterAcceptsRow(self, sourceRow, sourceParent):
+        index0 = self.sourceModel().index(sourceRow, 0, sourceParent)
+        index1 = self.sourceModel().index(sourceRow, 1, sourceParent)
+        return ((self.filterRegExp().indexIn(self.sourceModel().data(index0, self.role)) >= 0 or self.filterRegExp().indexIn(self.sourceModel().data(index1, self.role)) >= 0))
+
+class CCSearch(QWidget):
+    def __init__(self):
+        super().__init__()
+
+        self.resize(1280, 720)
+        self.filter_proxy_model = CustomFilter()
+        
+        mainLayout = QVBoxLayout()
+        units = pullFromJson()
+
+        model = self.createUnitItems(units=units)
+
+        self.createFilteredModel(model=model)
+
+        table = self.createTable()
+        
+        search_field = QLineEdit()
```

Comment:
Still think this should be better off extracted so that the `__init__` function is consistent, but I'll keep this as a suggestion instead of a request for change